### DISCUSSION TITLE
Cant compile app with wagmi

### DISCUSSION THREAD

**Discussion by Borming** [wevm/wagmi#1681] [Q&A]
I have a simple React 17 app, but as soon i as import wagmi

import { configureChains, mainnet } from 'wagmi';

i get this error :

ERROR in ./node_modules/wagmi/dist/index.js 188:33
Module parse failed: Unexpected token (188:33)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|   if (defaultedOptions.suspense && result.isLoading && result.isFetching && !isRestoring) {
|     throw observer.fetchOptimistic(defaultedOptions).then(({ data }) => {
>       defaultedOptions.onSuccess?.(data);
|       defaultedOptions.onSettled?.(data, null);
|     }).catch((error) => {
ℹ ｢wdm｣: Failed to compile.

Anyone knows whats going on?

**Comment by tmm** [comment]
What build tool are you using? The error message sounds like you need to target a more recent version of JavaScrip (e.g. with Babel).

**Reply by Borming** [reply]
Nah, i have all the latest versions of Babel, i am using Webpack 4 tho, but i dont think that should be a problem.

**Reply by tmm** [reply]
Definitely looks like a Webpack error related to optional chaining and not something on the wagmi side. I would poke around your Webpack config a bit more and double check that it supports optional chaining.

**Reply by Borming** [reply]
Thank, was thinking the same, but i have it in Webpack using Babel. https://babeljs.io/docs/en/babel-plugin-proposal-optional-chaining, i am using Webpack 4 with latest Babel @babel/preset-env. I used hundreds of libs and never had this problem.

Is Wagmi only for Webpack 5?

**Reply by tmm** [reply]
The Webpack version shouldn't matter, wagmi targets ECMAScript 2021 or newer so as long as you meet that it should be fine. Feel free to open an issue if you can come up with a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example).
